test/naptime/layers/dsl_test.clj

```diff
@@ -0,0 +1,22 @@
+(ns naptime.layers.dsl-test
+  (:require [ring.mock.request :as mock]
+            [clojure.test :refer :all]
+            [naptime.layers.dsl :as dsl]
+            [ring.middleware.params :as params]))
+
+(defn query-params [uri-with-query-string]
+  (-> (mock/request :get uri-with-query-string)
+      params/params-request
+      :query-params))
+
+;; test from this file: http://github.com/PostgREST/postgrest/blob/8e0255195615380ec18c46cb852e67dd199b5821/test/Feature/QuerySpec.hs#L33-L33
+(deftest dsl-tests
+  (testing "matches with equality"
+    (is (= [[[[:regular-column "id"]] [[:condition [:op [:eq [:number "5"]]]]]]]
+           (dsl/parse-params (query-params "items?id=eq.5")))))
+
+  (testing "matches with equality not using operator"
+    ;; TODO: not should not be a :column-name but :not
+    ;; parse-param return a bit too nesting; does it make sense to flatten a bit? does it matter when walking graph?
+    (is (= [[[[:regular-column "id"]] [[:condition [:column-name "not"] [:op [:eq [:number "5"]]]]]] [:order [[:column [:column-name "id"]]]]]
```

Comment:
notes to save time debugging

should have been `[:condition [:not "not"]` for some reason it is picking up as `:column-name` tried removing dot in the `<'not.'>`; doesn't matter

it wasn't because of defparser macro either i.e cached. I restarted REPL to test this